textile/features.textile

```diff
@@ -64,11 +64,45 @@ h2(#versions). Specification and Protocol Versions
+* @(G3)@ Testing statistics can be tricky due to timing issues and slow test suites as a result of sending requests to generate statistics.  As such, we provide a special stats endpoint in our sandbox environment that allows stats to be injected into our metrics system so that stats tests can make predictable assertions.  To create stats you must send an authenticated @POST@ request to the stats JSON to @https://sandbox.realtime.ably-nonprod.net/stats@ with the stats data you wish to create. See the "JavaScript stats fixture":https://github.com/ably/ably-js/blob/4e65d4e13eb8750a375b9511e4dd059092c0e481/spec/rest/stats.test.js#L8-L51 and "setup helper":https://github.com/ably/ably-js/blob/4e65d4e13eb8750a375b9511e4dd059092c0e481/spec/common/modules/testapp_manager.js#L158-L182 as an example
 * @(G4)@ This clause has been replaced by "@CSV1@":#CSV1 and "@CSV2@":#CSV2. It was valid up to and including specification version 1.2.
 ** @(G4a)@ This clause has been replaced by "@CSV2d@":#CSV2d. It was valid up to and including specification version 1.2.
 
+h2(#endpoint-configuration). Client library endpoint configuration
+
+* @(REC1)@ Various client options collectively determine a @primary domain@.
+** @(REC1a)@ The @primary domain@ is @main.realtime.ably.net@ unless overridden by specifying an @endpoint@ option or any of the deprecated options @environment@, @restHost@, @realtimeHost@.
+** @(REC1b)@ If the @endpoint@ option is specified then:
+*** @(REC1b1)@ If any one of the deprecated options @environment@, @restHost@, @realtimeHost@ are also specified then the options as a set are invalid.
+*** @(REC1b2)@ If the @endpoint@ option is a domain name, determined by it containing at least one period (@.@) then the @primary domain@ is the value of the @endpoint@ option.
+*** @(REC1b3)@ Otherwise, if the @endpoint@ option is a non-production routing policy name of the form @nonprod:[name]@ then the @primary domain@ is @[name].realtime.ably-nonprod.net@.
+*** @(REC1b4)@ Otherwise, if the @endpoint@ option is a production routing policy name of the form @[name]@ then the @primary domain@ is @[name].realtime.ably.net@.
+** @(REC1c)@ If the deprecated @environment@ option is specified then:
+*** @(REC1c1)@ If any one of the deprecated options @restHost@, @realtimeHost@ are also specified then the options as a set are invalid.
+*** @(REC1c2)@ If the @environment@ option is a routing policy name of the form @[name]@ then the primary domain is @[name].realtime.ably.net@.
+** @(REC1d)@ If either the @restHost@ or @realtimeHost@ option is specified then:
+*** @(REC1d2)@ If the @realtimeHost@ option is specified without also specifying @restHost@ then the options as a set are invalid.
+*** @(REC1d3)@ The @primary domain@ for REST operations is the value of the @restHost@ option.
+*** @(REC1d4)@ The @primary domain@ for Realtime operations is the value of the @realtimeHost@ option.
```

Comment:
That this is the only option that requires a library to keep a different domain for rest vs realtime operations. Is there a reason to have libs still bother to make this distinction? if not, couldn't we just say "Else if either `restHost` or `realtimeHost` are specified then the `primary domain` is `restHost || realtimeHost`" or something, and let the libs make that simplification?

Reply:
Sure, if you're confident that nobody's using different values for a valid reason. https://github.com/ably/specification/pull/213/commits/27a16960b674d44ba91873e22f3203756e7d7900